DigitalLearningSolutions.Web/Controllers/Register/RegisterController.cs

```diff
@@ -311,24 +321,32 @@ public async Task<IActionResult> Summary(SummaryViewModel model)
         }
 
         [HttpGet]
-        public IActionResult Confirmation()
+        public IActionResult Confirmation(
+            int? centreId,
+            string candidateNumber,
+            bool approved,
+            string? unverifiedPrimaryEmail,
+            string? centreSpecificEmail
+        )
         {
-            var candidateNumber = (string?)TempData.Peek("candidateNumber");
-            var approvedNullable = (bool?)TempData.Peek("approved");
-            var centreIdNullable = (int?)TempData.Peek("centreId");
-            TempData.Clear();
-
-            if (candidateNumber == null || approvedNullable == null || centreIdNullable == null)
+            if (centreId == null)
             {
                 return RedirectToAction("Index");
             }
 
-            var approved = (bool)approvedNullable;
-            var centreId = (int)centreIdNullable;
+            var centreIdForContactInformation = approved ? null : centreId;

```

Comment:
I'd expect this line to be in the view...

Reply:
This is just the centreId used for getting the contact info, looks like the actual info is retrieved from the db in the CentreContactInfoViewComponent

Reply:
What I mean is that we may as well pass the centreId to the view. The controller shouldn't be deciding when the view needs the centreId - that's the view's responsibility.